Applications/Spire/Source/Ui/TableHeaderItem.cpp

```diff
@@ -91,14 +93,50 @@ namespace {
     enclose(*sash, *resize_handle, Qt::AlignRight);
     return sash;
   }
+
+  class HeaderNameListModel : public ListModel<QString> {
```

Comment:
This is a step in the right direction, but the ```HeaderNameListModel``` isn't being updated when the ```source``` updates.